corehq/apps/api/resources/v0_5.py

```diff
@@ -845,9 +849,44 @@ def obj_get_list(self, bundle, **kwargs):
         if not application:
             return []
         forms_objects = application.get_forms(bare=False)
+
         for form_object in forms_objects:
             form = form_object['form']
             module = form_object['module']
             form_name = '{} > {} > {}'.format(application.name, module.name['en'], form.name['en'])
             results.append(Form(form_xmlns=form.xmlns, form_name=form_name))
         return results
+
+#Zapier requires id and name; case_type has no obvious id, placeholder inserted instead.s
+Case = namedtuple('Case', 'case_type placeholder')
+Case.__new__.__defaults__ = ('', '')
+
+
+class DomainCases(Resource):
+    """
+    Returns: list of case types for a domain
+
+    Note: only returns case types for which at least one case has been made
+    """
+    placeholder = fields.CharField(attribute='placeholder')
+    case_type = fields.CharField(attribute='case_type')
+
+    class Meta:
+        resource_name = 'domain_cases'
+        authentication = ApiKeyAuthentication()
+        object_class = Case
+        include_resource_uri = False
+
+    def obj_get_list(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        couch_user = CouchUser.from_django_user(bundle.request.user)
+        if not domain_has_privilege(domain, privileges.ZAPIER_INTEGRATION) or not couch_user.is_member_of(domain):
+            raise ImmediateHttpResponse(
+                HttpForbidden('You are not allowed to get list of cases for this domain')
```

Comment:
One option would be to use the resource's meta class to control authorization: https://django-tastypie.readthedocs.io/en/latest/authorization.html